### PR TITLE
refactor(inline): range sync

### PR DESCRIPTION
**Pull request by Flrande**
Close [BS-1332](https://linear.app/affine-design/issue/BS-1332/latex-editor-输入光标不正常)

### CODE REVIEW

**Review by doodlewind — COMMENTED** [review]
Wondering are there some RFCs or descriptions about the refactoring?

**Comment by Flrande** [comment]
> Wondering are there some RFCs or descriptions about the refactoring?

No, because this PR did not cause many API changes; it mainly focuses on optimizing the existing code.

**Review by Saul-Mirone — CHANGES_REQUESTED** [review]
Waiting for resolving IME issues.

File: packages/framework/inline/src/services/render.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  };
+
+  rerenderWholeEditor = () => {
+    const rootElement = this.editor.rootElement;
+
+    if (!rootElement.isConnected) return;
+
+    rootElement.replaceChildren();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    delete (rootElement as any)['_$litPart$'];
```
Comment (Saul-Mirone) [inline packages/framework/inline/src/services/render.ts:159]: Need to add a comment to explain what is this.

File: packages/framework/block-std/src/range/range-binding.ts [hunk 2]
```diff
@@ -130,23 +123,18 @@ export class RangeBinding {
...
-
-        // Restore the DOM structure damaged by the composition
-        parent.dirty = true;
-        await parent.updateComplete;
-        await parent.updateComplete;
-        parents.push(parent);
-      }
+      this.host.renderRoot.replaceChildren();
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      delete (this.host.renderRoot as any)['_$litPart$'];
```
Comment (Saul-Mirone) [inline packages/framework/block-std/src/range/range-binding.ts:128]: Same here

**Review by Saul-Mirone — APPROVED** [review]

**Pull request merged**